storage/allocator.go

```diff
@@ -13,75 +13,291 @@
 // permissions and limitations under the License. See the AUTHORS file
 // for names of contributors.
 //
-// Author: Levon Lloyd (levon.lloyd@gmail.com)
 // Author: Spencer Kimball (spencer.kimball@gmail.com)
+// Author: Kathy Spradlin (kathyspradlin@gmail.com)
 
 package storage
 
 import (
+	"fmt"
 	"math/rand"
+	"sort"
+	"sync"
 
+	"github.com/cockroachdb/cockroach/gossip"
 	"github.com/cockroachdb/cockroach/proto"
 	"github.com/cockroachdb/cockroach/util"
 )
 
-// allocator makes allocation decisions based on a zone configuration,
-// existing range metadata and available stores. Configuration
-// settings and range metadata information is stored directly in the
-// engine-backed range they describe. Information on suitability and
-// availability of servers is gleaned from the gossip network.
+const (
+	// minAvailCapacityThreshold: if the capacity of a store descriptor
+	// falls below this percentage, it will never be used as an
```

Comment:
Nit: this is a fraction, not a percentage. I might rename percent/percentage to fraction everywhere to make this clearer.